lib/galaxy/jobs/__init__.py

```diff
@@ -1402,6 +1429,7 @@ def _set_object_store_ids(self, job):
         # afterward. State below needs to happen the same way.
         for dataset_assoc in job.output_datasets + job.output_library_datasets:
             dataset = dataset_assoc.dataset
+            self.__assign_media(job, dataset.dataset)
```

Comment:
I've spent months of my life trying to optimize this process of initializing the output datasets. Can we have some property on app that we can check to see if this method would ever doing anything - and skip it if there is no possibility of assigning media?

Reply:
Sure! [this config property](https://github.com/galaxyproject/galaxy/pull/4840/files#diff-de4bb10f1af2a1d9d642c3904b86818cR1112) is added that [if disabled, this method will not do anything](https://github.com/galaxyproject/galaxy/pull/4840/files#diff-3c62e82e72ce988e633066c5a9b670bbR976). Would that be addressing your concerns?